x-pack/plugins/stack_connectors/server/connector_types/microsoft_defender_endpoint/o_auth_token_manager.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+import { ConnectorUsageCollector } from '@kbn/actions-plugin/server/usage';
+import { MicrosoftDefenderEndpointDoNotValidateResponseSchema } from '../../../common/microsoft_defender_endpoint/schema';
+import {
+  MicrosoftDefenderEndpointConfig,
+  MicrosoftDefenderEndpointSecrets,
+  MicrosoftDefenderEndpointApiTokenResponse,
+} from '../../../common/microsoft_defender_endpoint/types';
+
+export class OAuthTokenManager {
```

Comment:
I see you introduced a custom Token class 👍 , have you tried the one from Response Ops? Does it meet our needs?